module/VuFind/tests/unit-tests/src/VuFindTest/ILS/Logic/HoldsTest.php

```diff
@@ -98,4 +101,322 @@ public function testGetSuppressedLocations(array $configArray, array $expectedLi
         $logic = $this->getHoldsLogic(config: $configArray);
         $this->assertEquals($expectedList, $logic->getSuppressedLocations());
     }
+
+    /**
+     * Test formatHoldings method
+     *
+     * @return void
+     */
+    public function testFormatHoldings(): void
+    {
+        $catalog = $this->createMock(Connection::class);
+        $catalog->method('getHoldingsTextFieldNames')->willReturn(['notes', 'summary']);
+
+        $logic = $this->getHoldsLogic(catalog: $catalog);
+
+        // Use reflection to access the protected formatHoldings method
+        $reflection = new \ReflectionClass($logic);
+        $method = $reflection->getMethod('formatHoldings');
+        $method->setAccessible(true);
+
+        // Example holdings data
+        // note that this is a simplified example; actual data may vary and include an availability object
+        $holdings = $this->getJsonFixture('ils/holdings_example.json');
+
+        $result = $method->invoke($logic, $holdings);
+
+        // assert various properties of the result
+        // I need to update the fixture and test to be more readable - remove UUIDs and replace with simple strings
+        $this->assertArrayHasKey('location1_id|Main Library', $result);
+        $this->assertEquals('Main Library', $result['location1_id|Main Library']['location']);
+        $this->assertCount(2, $result['location1_id|Main Library']['items']);
+        $this->assertEquals('02.09.2025', $result['location1_id|Main Library']['items'][0]['duedate']);
+        $this->assertEquals('Circulating', $result['location1_id|Main Library']['items'][0]['loan_type_name']);
+        $this->assertEquals(true, $result['location1_id|Main Library']['items'][0]['is_holdable']);
+        $this->assertEquals(
+            'item_id_1',
+            $result['location1_id|Main Library']['items'][0]['item_id']
+        );
+        $this->assertEquals(
+            'holdings_id_1',
+            $result['location1_id|Main Library']['items'][0]['holdings_id']
+        );
+        $this->assertEquals(
+            'item_id=item_id_1&holdings_id=holdings_id_1&status=Checked+out&hashKey=sample_hash_key',
+            $result['location1_id|Main Library']['items'][0]['link']['query']
+        );
+    }
+
+    /**
+     * Create a mock availability status for testing
+     *
+     * @param bool   $available   Whether the item is available
+     * @param string $description Status description
+     *
+     * @return object
+     */
+    protected function createMockAvailabilityStatus($available = true, $description = 'Available')
+    {
+        // Create an anonymous class that implements the required methods
+        // generated with Claude Sonnet 4 - but there must be a better way to do this,
+        // i.e. by hooking into the AvailabilityStatusInterface directly.
+        return new class ($available, $description) {
```

Comment:
Is there a reason you can't just use (and/or partially mock) the existing [AvailabilityStatus](https://github.com/vufind-org/vufind/blob/dev/module/VuFind/src/VuFind/ILS/Logic/AvailabilityStatus.php) class? I'm not sure exactly why an anonymous class is needed here.

Reply:
The truth is, that my experience with Unit Tests in PHP is sadly very limited and I asked AI for help - sorry - hanging head in shame :(

Reply:
Please let me know if there's anything I can help clear up for you! Happy to have a chat on Slack (or here, or anywhere else convenient) if that would be useful.

Reply:
I've made changes to this and provided a commit that hooks straight into \VuFind\ILS\Logic\AvailabilityStatus - let's hope this makes it better.